src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -305,6 +308,9 @@ public void Reset()
 
             _manuallySetRequestAbortToken = null;
             _abortedCts = null;
+
+            _maxRequestHeadersTotalSize = ServerOptions.Limits.MaxRequestHeadersTotalSize;
```

Comment:
nit: `_remainingRequestHeadersBytesAllowed` or something like that since it changes as headers come in.